src/Microsoft.ML.Featurizers/TimeSeriesImputer.cs

```diff
@@ -637,6 +650,33 @@ internal override byte[] GetSerializedValue()
             }
         }
 
+        private class DateTimeTypedColumn : TypedColumn<DateTime>
+        {
+            private static readonly DateTime _unixEpoch = new DateTime(1970, 1, 1);
+            private readonly bool _isNullable;
+
+            internal DateTimeTypedColumn(DataViewSchema.Column column, bool isNullable = false) :
+                base(column)
+            {
+                _isNullable = isNullable;
+            }
+
+            internal override byte[] GetSerializedValue()
+            {
+                var dateTime = GetValue();
+                byte[] bytes;
+
+                var value = dateTime.Subtract(_unixEpoch).Ticks / TimeSpan.TicksPerSecond;
+
+                bytes = BitConverter.GetBytes(value);
```

Comment:
This is going to allocate a new, very small, `byte[]` every time it is called. That is going to make the garbage collector do a lot of extra work to clean these small arrays up. Have we done perf testing on this?

Maybe a better design here would be to ask the Column for the size it needs, and then pass an appropriately sized `Span<byte>` into the Column, telling it to write to that Span. This would use as few allocations as necessary.

Reply:
It would also avoid having to then copy this data into a different buffer that gets passed into native code. Instead, the `TypedColumn` could just write to the "final" place directly.

---
In reply to: [377914670](https://github.com/dotnet/machinelearning/pull/4812#discussion_r377914670) [](ancestors = 377914670)

Reply:
Alright, I changed the code to use a `MemoryStream` and then I reuse the underlying `byte[]` stream for each call. The `MemoryStream` handles the resizing if needed so I removed that from my code, and now it should only be creating the `byte[]` once directly using the `MemoryStream`.